src/reactions/Reaction.js

```diff
@@ -26,24 +28,79 @@ const styles = StyleSheet.create({
   frameNotVoted: {
     borderColor: '#dedede', // eslint-disable-line
   },
-  countCommon: {
+  placeHolderCount: {
     marginLeft: 4,
+    color: 'transparent'
   },
   countVoted: {
     color: BRAND_COLOR,
   },
   countNotVoted: {
     color: 'gray',
   },
+  roller: {
+    position: 'absolute',
+    top: -20,
+    right: 4,
+  },
+  rollerText: {
+    paddingTop: 3,
+    paddingBottom: 3
+  }
 });
 
+const forwardAnimationConfig = {
+  fromValue: 0,
+  toValue: 1,
+  duration: 400,
+  easing: Easing.bezier(0.17, 0.67, 0.11, 0.99)
+};
+
+const backAnimationConfig = {
+  ...forwardAnimationConfig,
+  fromValue: 1,
+  toValue: 2
+};
+
 class Reaction extends React.PureComponent {
 
+  state = {
+    voteChangeAnimation: new Animated.Value(0)
+  };
+
   props: {
     name: string,
     voted: boolean,
   };
 
+  componentWillReceiveProps = (nextProps) => {
+    const { voteCount } = this.props;
+
+    if (nextProps.voteCount > voteCount) {
+      this.incrementCounter();
+    } else if (nextProps.voteCount < voteCount) {
+      this.decrementCounter();
+    }
+  }
+
+  animation = (fn, stateVar, config, reverse) => {
```

Comment:
Make the function more specific. That is a bit of `premature generalization`.

`fn`, `stateVar` and `config` never change. Just inline the values and it will make the code more clear by allowing the reader to not look in multiple places.